controllers/user.controller.js

```diff
@@ -0,0 +1,37 @@
+const userServices = require("../services/user.services");
+const { userSchema } = require("../schemas/user");
+const { resolveError } = require("../helpers/controllers");
+
+const list = async (req, res) => {
+  try {
+    const users = await userServices.list(req.query);
+    return res.status(200).json(users);
+  } catch (err) {
+    return resolveError(err, res);
+  }
+};
+
+const create = async (req, res) => {
+  try {
+    const value = await userSchema.validateAsync(req.body);
+    const user = await userServices.create(value);
+    return res.status(201).json({ id: user.id });
+  } catch (err) {
+    return resolveError(err, res);
+  }
+};
+
+const update = async (req, res) => {
+  try {
+    const data = req.body;
```

Comment:
where is data validation here?

Reply:
how do i validate this if i can only change role and isActive? do i just validate these two attributes?

Reply:
create a new schema validation